File: Segundo-Bimestre/Parcial 1/Funciones_Procedimientos/Funciones_Procedimientos_java/src/funciones_procedimientos/Funcion4_2B.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package funciones_procedimientos;
import java.util.Scanner;
import static funciones_procedimientos.Funcion3_2B.Impar;
/**
 *
 * @author DELL
 */
public class Funcion4_2B {
    public static void main(String[] args) {
        Scanner leer = new Scanner (System.in);
        
        int num = 0;
        boolean esImpar, esPrimo;
        
        System.out.println("Función para verificar si un número es impar y primo");
        
        System.out.print("Ingrese el número a verificar: ");
        num = leer.nextInt();
        
        esImpar = Impar(num);
        
        if (esImpar == true) {
            esPrimo = Primo(num);
            
            if (esPrimo == true) {
                System.out.println("El número "+num+" es impar y es primo");
            }
            
            else {
                System.out.println("El número "+num+" es impar pero no es primo");
            }
        }
        else {
            System.out.println("El número "+num+" no es impar, por lo tanto no se verifica que sea primo");
        }
    }
    
    public static boolean Primo (int numero) {
        int div = 0;
        
        for (int i=1; i<=numero; i++) {
            if (numero % i == 0) {
                div = div + 1;
            }
        }
        
        if (div == 2) {
            return true;
        }
        
        else {
            return false;
        }
    }
}

File: Segundo-Bimestre/Parcial 1/Funciones_Procedimientos/Funciones_Procedimientos_pyhton/Funcion4_2B.py
print("")
# ****Método principal*****

# Inicializar las variables
num = 0
esImpar = True and False
esPrimo = True and False

# Ingresar los datos
print("Función para verificar si un número es impar y primo")
num = int(input("Ingrese el número a verificar: "))

# Función Primo
def Primo (numero):
    div = 0

    for i in range (1, numero+1):
        if (numero % i == 0):
            div = div + 1
    
    if (div == 2):
        return True

    else: 
        return False

# Función Impar
def Impar(numero):
    if (numero % 2 == 0):
        return False
    else:
        return True

# Asignarle un valor a la función
esImpar = Impar(num)

# Proceso
if (esImpar == True):
    esPrimo = Primo(num)

    if (esPrimo == True):
        print("El número ",num," es impar y es primo")
    
    else:
        print("El número ",num," es impar pero no es primo")

else:
    print("El número ",num," no es impar, por lo tanto no se verifica que sea primo")